api_app/script_analyzers/observable_analyzers/triage_search.py

```diff
@@ -0,0 +1,122 @@
+import time
+import logging
+import requests
+
+from api_app.exceptions import AnalyzerRunException
+from api_app.script_analyzers import classes
+from intel_owl import secrets
+
+
+logger = logging.getLogger(__name__)
+
+
+class TriageSearch(classes.ObservableAnalyzer):
+    # using public endpoint as the default url
+    base_url: str = "https://api.tria.ge/v0/"
+    private_url: str = "https://private.tria.ge/api/v0/"
+
+    def set_config(self, additional_config_params):
+        self.endpoint = additional_config_params.get("endpoint", "public")
+        if self.endpoint == "private":
+            self.base_url = self.private_url
+
+        self.api_key_name = additional_config_params.get("api_key_name", "TRIAGE_KEY")
+        self.__api_key = secrets.get_secret(self.api_key_name)
+        self.analysis_type = additional_config_params.get("analysis_type", "search")
+        self.report_type = additional_config_params.get("report_type", "overview")
+        self.max_tries = additional_config_params.get("max_tries", 1)
+        self.poll_distance = 3
+
+    def run(self):
+        if not self.__api_key:
+            raise AnalyzerRunException(
```

Comment:
should be [`AnalyzerConfigurationException`](https://github.com/intelowlproject/IntelOwl/blob/fa472db8c9b3d4c7ec5201ec552f6f928b076814/api_app/exceptions.py#L5)